napari/layers/tracks/tracks.py

```diff
@@ -496,7 +496,8 @@ def _recolor_tracks(self):
         # updated before the properties are. properties should always contain
         # a track_id key
         if self.color_by not in self.properties_to_color_by:
-            self.color_by = 'track_id'
+            self._color_by = 'track_id'
+            self.events.color_by()
```

Comment:
Is this event required? Is it there because the layer controls combobox isn't updated without it?

Reply:
I inserted this line to reproduce the color_by setter behavior but without recoloring.
It should update the ComboBox because it calls the color_by EventEmitter, but TBH I haven't completely understood napari's Events.

Reply:
> I inserted this line to reproduce the color_by setter behavior but without recoloring.
It should update the ComboBox because it calls the color_by EventEmitter,

This makes sense to me. Thanks for the explanation,.

>  but TBH I haven't completely understood napari's Events.

Ha, yeah, I also get confused by them sometimes 🤷 . Hopefully we can simplify them in the future!